### PR TITLE
Update bitvec to 1.0.0

### PR DESCRIPTION
**Pull request by jmcook1186**
PR updates BitVec version to 1.0.0 and fixes bugs arising from this change.

All tests pass.

### CODE REVIEW

File: ssz_rs/src/merkleization/cache.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ impl Cache {
     pub fn with_leaves(leaf_count: usize) -> Self {
         Self {
             leaf_count,
-            dirty_leaves: bitvec![1; leaf_count],
+            dirty_leaves: bitvec![1,],
```
Comment (ralexstokes) [inline ssz_rs/src/merkleization/cache.rs:15]: does this make a bitvector of length `leaf_count` that has all bits set?

**Review by ralexstokes — CHANGES_REQUESTED** [review]
nice work!

from scanning the commits, it looks like we only need d91c0e3f087cfb2c2ddca011693612e191e3df26 in this PR

can you rebase this branch to drop the other 3?

File: ssz_rs/src/merkleization/cache.rs (hunk 1 above)
Comment (jmcook1186) [reply ssz_rs/src/merkleization/cache.rs]: Now updated (055129e) to explicitly define bitvec length to be equal to leaf_count.

**Comment by jmcook1186** [comment]
Rebased to remove uneccessary commits. 

New commit revises definition of `dirty_leaves`

BitVec 0.22 -> BitVec 1.0.0 seems to include a breaking change for how `bitvec!` macro works in the Cache definition (`/merkleize/cache.rs` line 13).

The original way now throws an error because the macro expects a constant and we need to pass in a variable (`leaf_count`).

I couldn't find a more elegant solution to this other than moving the bitvec construction to precede the struct definition.

I also included two `assert_eq's` to demonstrate the length and content are OK. These can be removed later.

**Pull request closed without merging**